### DISCUSSION TITLE
SRM Notes for 6-11-2024

### DISCUSSION THREAD

**Discussion by bruno-f-cruz** [harp-tech/protocol#105] [SRM Notes]
### Revisit:

https://github.com/bonsai-rx/roadmap/issues

- https://github.com/harp-tech/protocol/issues/41
- https://github.com/harp-tech/protocol/issues/22
- 
- https://github.com/harp-tech/protocol/issues/48
- https://github.com/harp-tech/protocol/pull/45
- https://github.com/harp-tech/protocol/pull/39
- https://github.com/harp-tech/protocol/pull/38

**Comment by glopesdev** [comment]
Re. https://github.com/harp-tech/protocol/issues/22 we have discussed briefly a summary of the current proposal and have started a draft implementation of the required changes in https://github.com/harp-tech/protocol/pull/57.

@bruno-f-cruz please feedback with anything I may be missing. One thing which remains to decide whether to include together in the same PR is a new field detailing the platform architecture.

Related to this, if we intend to split the platform minimal implementations, we might decide to host different `core.yml` files in the protocol repository, e.g. we could have a `atxmega.yml` with the current registers for ATXMega and a `rp2040.yml` with the implemented registers for the Pico, but this may again prove too much for us to keep track of in parallel and we might decide to simply support the closest superset of all registers.

**Reply by bruno-f-cruz** [reply]
Reviewed the PR. 

> but this may again prove too much for us to keep track of in parallel and we might decide to simply support the closest superset of all registers.

I am ok with this under the assumption that the superset is the union of all specs and that has, as an upper bound, the harp protocol spec. In other words, no registers should exist in the core.yml that are not merged to main in the protocol spec.
